src/status_im/contexts/wallet/send/events.cljs

```diff
@@ -556,7 +559,11 @@
               (assoc-in [:wallet :ui :send :just-completed-transaction?] true)
               (assoc-in [:wallet :transactions] transaction-details)
               (assoc-in [:wallet :ui :send :transaction-ids] transaction-ids))
-      :fx [[:dispatch
+      :fx [;; Remove wallet/stop-and-clean-suggested-routes event when we move to new transaction
+           ;; submission process as the routes as stopped by status-go
+           [:dispatch
+            [:wallet/stop-and-clean-suggested-routes]]
+           [:dispatch
```

Comment:
One thing came to mind when reading through the code: 
* What should we do when something fails?
  * For example, if the transaction fails, we may still be subscribing to suggested-routes because it seems we start listening for suggested-routes but only stop if the transaction succeeds.

Reply:
If the `createMultiTransaction` RPC throws an error, the user could see a toast message of the error. The user would still be on the tx confirmation page. We don't actually clean the data in app-db and allow the users to submit the transaction again if required. 

https://github.com/status-im/status-mobile/blob/f94be17b1133599aea32b9c6186448305833fb0e/src/status_im/contexts/wallet/send/events.cljs#L655-L663

If the user decides to go back (not performing the transaction), he/she will go back through the routes page, which will stop fetching the routes.

Reply:
Ah nice that makes sense, thank you for clarifying 🙏 

Is it possible for the user to cancel their transaction, by navigating backwards or something, and if they can/do will they still be subscribing to the suggested routes? I ask because I had noticed in the wallet-send flow for tokens, that we start subscribing but I'm not sure if we have an event for unsubscribing when cancelling. Thoughts?

Reply:
There is no explicit cancel transaction in our UI which closes the transaction flow completely (like in the desktop).
If the user decides not to perform the transaction after reaching tx confirmation page, he/she needs to navigate back. We would stop the route generation (cancelling subscription to the routes).

Also in collectible tx flow the route generation screen is skipped and the user is taken to the tx confirmation page. That's why we have this check to stop the routes when the user navigates back from the tx confirmation page.

https://github.com/status-im/status-mobile/pull/21306/files#diff-089fb8d72eefa455e3dc6437b0bc11c3ca3cc0fb4063e20f129b09929aeae3a9R730-R732

Let me know If that answers your question.

Reply:
Ah yes that makes sense! Nice stuff ✅